datafusion/src/logical_plan/builder.rs

```diff
@@ -291,6 +291,15 @@ impl LogicalPlanBuilder {
         Ok(Self::from(union_with_alias(self.plan.clone(), plan, None)?))
     }
 
+    /// apply union distinct
+    pub fn union_distinct(&self) -> Result<Self> {
```

Comment:
```suggestion
    /// Apply deduplication: Only distinct (different) values are returned)
    pub fn distinct(&self) -> Result<Self> {
```

I suggest renaming this to just `distinct` as there is only one input, so I find the use of the term `union` somewhat confusing